gr-digital/lib/meas_evm_cc_impl.cc

```diff
@@ -0,0 +1,78 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "meas_evm_cc_impl.h"
+#include <gnuradio/io_signature.h>
+
+namespace gr {
+namespace digital {
+
+meas_evm_cc::sptr meas_evm_cc::make(constellation_sptr cons, evm_measurement_t meas_type)
+{
+    return gnuradio::get_initial_sptr(new meas_evm_cc_impl(cons, meas_type));
+}
+
+meas_evm_cc_impl::meas_evm_cc_impl(constellation_sptr cons, evm_measurement_t meas_type)
+    : gr::sync_block("meas_evm_cc",
+                     gr::io_signature::make(1, 1, sizeof(gr_complex)),
+                     gr::io_signature::make(1, 1, sizeof(float))),
+      d_cons(cons),
+      d_meas_type(meas_type)
+{
+
+    d_cons_points = d_cons->points();
+
+    double sum = 0.0;
+    int N = d_cons_points.size();
+    for (int i = 0; i < N; i++) {
+        sum += pow(real(d_cons_points[i]), 2.0) + pow(imag(d_cons_points[i]), 2.0);
+    }
+    sum /= (double)N;
+    d_cons_mag_sq = sum;
+    d_cons_mag = sqrt(sum);
+}
+
+double meas_evm_cc_impl::mag(gr_complex x)
```

Comment:
`std::abs` does exactly this, so this feels kind of duplicate, unless we really need the squares and square root to happen on doubles instead of floats.

Reply:
generally, are we using this?